internal/credit/postgres_connector/grant_test.go

```diff
@@ -40,12 +42,53 @@ func TestPostgresConnectorGrants(t *testing.T) {
 		description string
 		test        func(t *testing.T, connector credit.Connector, db_client *db.Client, ledger credit.Ledger)
 	}{
+		{
+			name:        "Should truncate grant effectiveAt to windowsize",
+			description: "EffectiveAt should be the end date of the window the input falls into",
+			test: func(t *testing.T, connector credit.Connector, db_client *db.Client, ledger credit.Ledger) {
+				ctx := context.Background()
```

Comment:
I would also add a testcase here if the effective time is already truncated. The code for truncate is good, just I would ensure that if we sometimes touch that codebase then we won't ruin the current logic.

Reply:
That happens in the default create testcase (there it's truncated)